modules/catalog/src/test/java/org/apache/ignite/internal/catalog/serialization/CatalogEntrySerializationCompatibilityTest.java

```diff
@@ -0,0 +1,429 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.catalog.serialization;
+
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.assertThrows;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.ignite.internal.catalog.Catalog;
+import org.apache.ignite.internal.catalog.storage.UpdateEntry;
+import org.apache.ignite.internal.catalog.storage.VersionedUpdate;
+import org.apache.ignite.internal.util.io.IgniteDataInput;
+import org.apache.ignite.internal.util.io.IgniteDataOutput;
+import org.apache.ignite.lang.MarshallerException;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests to verify catalog storage entries serialization compatibility.
+ */
+public class CatalogEntrySerializationCompatibilityTest {
+    private final CatalogEntrySerializerProvider serializerProvider = (typeId) -> {
```

Comment:
I think 2 versions is enough. The test must checks next cases:
* Marshaller of version X, must successfully marshal-unmarshal an entry.
* Marshaller of version 2, must successfully unmarshal an entry marshalled by marshaller version 1.
* Marshaller of version 1, must fails unmarshal an entry marshalled by marshaller version 2.

Reply:
Done, test simplified, removed 3rd version of entry and serializer.

```
Marshaller of version X, must successfully marshal-unmarshal an entry.
Marshaller of version 2, must successfully unmarshal an entry marshalled by marshaller version 1.
```
implemented in `readEntryInPreviousFormat`

```
Marshaller of version X, must successfully marshal-unmarshal an entry.
Marshaller of version 1, must fails unmarshal an entry marshalled by marshaller version 2.
```
implemented in `forwardCompatibilityIsNotSupported`

```
Serializer of version 2 can *write* entry in version 1 format
```
implemented in `writeEntryInPreviousFormat`